### ISSUE TITLE
Generate test function from a test file results in error opening file

### ISSUE THREAD

**Issue by pot-code** [microsoft/vscode-go#2822]
Steps to Reproduce:

1. create a test file like `xxx_test.go`
2. move cursor to a function name(not a function prefixed with Test) and generate a test function, a error message then appears in the corner

What impressed me is that the desired test function is appended to file, without overwriting the current content.

Is there any way to remove the error message? Every time I generate a test function, it will appear.

Though it may be uncommon to do it, but it's useful sometimes, would you mind supporting it?

**Comment by ramya-rao-a** [comment]
Thanks for reporting @pot-code, can you share the error message you are seeing?

**Comment by pot-code** [comment]
Created a file named `demo_test.go` **without** the associated `demo.go` file

```
Unable to open 'demo.go': Unable to read file (Error: File not found (/Users/xxx/Documents/sandbox/playground/go/src/xxx.lang/spec/demo.go)).
```

It seems that it tries to open the `demo.go` file after creating the test function

**Comment by ramya-rao-a** [comment]
Ah! understood.

The assumption always was that the "generate unit tests" command would be run on a non test file.
So, when the corresponding test file was updated, there was a need to open it.
For this, I re-used the `Toggle Test File` command.

In your case, the `Toggle Test File` command fails because the non test file doesnt exist :)

PRs are welcome to fix this bug
- [Set up local dev env](https://github.com/microsoft/vscode-go/wiki/Building,-Debugging-and-Sideloading-the-extension-in-Visual-Studio-Code)
- https://github.com/microsoft/vscode-go/blob/master/src/goGenerateTests.ts#L176 is where we attempt to toggle the file. You will have to add a check to not do this if we are operating on a test file already
- You can pass a flag via the [config passed to the generateTests](https://github.com/microsoft/vscode-go/blob/master/src/goGenerateTests.ts#L121) function to determine if the file from where the command was run was a test file or not

**Comment by drakegens** [comment]
I will try to take this one on!

**Comment by aykejriw** [comment]
@ramya-rao-a If no one is working here, I can do that.

**Comment by ramya-rao-a** [comment]
I believe @drakegens has a PR out for this that had some feedback they had to react to.
@drakegens Are you planning on continuing your work in #2883?

**Comment by drakegens** [comment]
Yes I will continue my work. Sorry work has been crazy.




On Mon, Nov 18, 2019 at 10:44 PM -0600, "Ramya Rao" <notifications@github.com<mailto:notifications@github.com>> wrote:


I believe @drakegens<https://github.com/drakegens> has a PR out for this that had some feedback they had to react to.
@drakegens<https://github.com/drakegens> Are you planning on continuing your work in #2883<https://github.com/microsoft/vscode-go/pull/2883>?

**Comment by ramya-rao-a** [comment]
This issue is fixed in the latest version (0.13.0) of this extension

Thanks for the PR @drakegens and Thanks for reporting @pot-code !

Happy Coding!
